indico/modules/designer/util.py

```diff
@@ -42,6 +45,21 @@ def get_inherited_templates(obj):
     return get_all_templates(obj) - set(obj.designer_templates)
 
 
+def get_not_deletable_templates(obj):
+    """Get all not deletable templates usable by an event/category"""
+
+    not_deletable_criteria = [
+        DesignerTemplate.is_system_template,
+        DesignerTemplate.backside_template_of != None,
+        DesignerTemplate.ticket_for_regforms.any(RegistrationForm.event_new.has(Event.ends_after(now_utc())))
+    ]
+    not_deletable = set(DesignerTemplate.query
+                        .filter(DesignerTemplate.event_new == obj if isinstance(obj, Event) else
+                                DesignerTemplate.category_id.in_(categ['id'] for categ in obj.chain))
+                        .filter(db.or_(*not_deletable_criteria)))
+    return not_deletable
```

Comment:
why not `return set(...)` directly instead of assigning it first just to return it?